### PR TITLE
Verify the client version.

### PR DESCRIPTION
**Pull request by CPWstatic**
nebua python client had been updated. https://github.com/vesoft-inc/nebula-python/pull/148

### CODE REVIEW

File: src/graph/service/GraphService.cpp [hunk 1]
```diff
@@ -175,5 +175,22 @@ bool GraphService::auth(const std::string& username, const std::string& password
...
 }
 
+folly::Future<cpp2::VerifyClientVersionResp> GraphService::future_verifyClientVersion(
+    const cpp2::VerifyClientVersionReq& req) {
+  std::unordered_set<std::string> whiteList;
+  folly::splitTo<std::string>(
+      ":", FLAGS_client_white_list, std::inserter(whiteList, whiteList.begin()));
+  cpp2::VerifyClientVersionResp resp;
+  if (FLAGS_enable_client_white_list && whiteList.find(req.get_version()) == whiteList.end()) {
+    resp.set_error_code(nebula::cpp2::ErrorCode::E_CLIENT_REJECTED);
```
Comment (Shylock-Hg) [inline src/graph/service/GraphService.cpp:185]: E_CLIENT_INCOMPATIBLE is more clear.

File: src/graph/service/GraphFlags.cpp [hunk 2]
```diff
@@ -65,3 +67,8 @@ DEFINE_bool(disable_octal_escape_char,
             " in next version to ensure compatibility with cypher.");
 
 DEFINE_bool(enable_experimental_feature, false, "Whether to enable experimental feature");
+
+DEFINE_bool(enable_client_white_list, true, "Turn on/off the client white list.");
+DEFINE_string(client_white_list,
+              nebula::getOriginVersion() + ":2.5.0:2.5.1",
+              "A white list for diffrent client versions, seperate with colon.");
```
Comment (yixinglu) [inline src/graph/service/GraphFlags.cpp:74]: different

File: src/interface/common.thrift [hunk 3]
```diff
@@ -33,6 +33,8 @@ cpp_include "common/datatypes/HostAddrOps-inl.h"
  *
  */
 
+const binary (cpp.type = "char const *") version = "2.5.1"
```
Comment (yixinglu) [inline src/interface/common.thrift:36]: 2.6.0

**Review by yixinglu — COMMENTED** [review]
Our clients need to verify the version compatibility

File: src/interface/graph.thrift [hunk 4]
```diff
@@ -99,6 +99,17 @@ struct AuthResponse {
...
 
 
+struct VerifyClientVersionResp {
+    1: required common.ErrorCode error_code;
+    2: optional binary           error_msg;
+}
+
+
+struct VerifyClientVersionReq {
+    1: required binary version = common.version;
```
Comment (Shylock-Hg) [inline src/interface/graph.thrift:109]: This field should provided by client. Why set default value by server binary?

**Review by yixinglu — DISMISSED** [review]
LGTM

**Review by Shylock-Hg — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]

**Pull request merged**